util/progress/metricwriter.go

```diff
@@ -446,14 +448,22 @@ func newLintMetricRecorder(meter metric.Meter, attrs attribute.Set) *lintMetricR
 	return mr
 }
 
+func kebabToCamel(s string) string {
+	words := strings.Split(s, "-")
+	for i, word := range words {
+		words[i] = cases.Title(language.English).String(word)
+	}
+	return strings.Join(words, "")
+}
+
 func (mr *lintMetricRecorder) Record(ss *client.SolveStatus) {
 	for _, warning := range ss.Warnings {
-		m := reLintMessage.FindSubmatch(warning.Short)
-		if m == nil {
+		m := reLintMessage.FindSubmatch([]byte(warning.URL))
+		if m == nil || len(m) < 2 {
```

Comment:
nit: technically `nil` check is not needed anymore

Reply:
Removed redundant nil!